--- ctypesgen/parser/preprocessor.py.orig
+++ ctypesgen/parser/preprocessor.py
@@ -159,10 +159,29 @@
 
         self.cparser.handle_status(cmd)
 
-        pp = subprocess.Popen(
-            cmd, shell=True, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-        )
-        ppout, pperr = pp.communicate()
+        pp = subprocess.Popen(cmd,
+                              shell=True,
+                              universal_newlines=True,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE)
+        try:
+            ppout, pperr = pp.communicate()
+        except UnicodeError:
+            # Fix for https://trac.osgeo.org/grass/ticket/3883,
+            # handling file(s) encoded with mac_roman
+            if sys.platform == 'darwin':
+                pp = subprocess.Popen(cmd,
+                                      shell=True,
+                                      universal_newlines=False,  # read as binary
+                                      stdout=subprocess.PIPE,
+                                      stderr=subprocess.PIPE)
+                ppout, pperr = pp.communicate()
+
+                data = ppout.decode('utf8', errors='replace')
+                ppout = data.replace('\r\n', '\n').replace('\r', '\n')
+                pperr = pperr.decode('utf8', errors='replace')
+            else:
+                raise UnicodeError
 
         for line in pperr.split("\n"):
             if line:
