diff --git a/lib/gis/gisinit.c b/lib/gis/gisinit.c
index c45a9eb15..1531d18da 100644
--- a/lib/gis/gisinit.c
+++ b/lib/gis/gisinit.c
@@ -49,12 +49,12 @@ void G__gisinit(const char *version, const char *pgm)
 
     G_set_program_name(pgm);
 
-    if (strcmp(version, GIS_H_VERSION) != 0)
+    /*if (strcmp(version, GIS_H_VERSION) != 0)
 	G_fatal_error(_("Module built against version %s but "
 			"trying to use version %s. "
 			"You need to rebuild GRASS GIS or untangle multiple installations."),
                         version, GIS_H_VERSION);
-    
+    */
     /* Make sure location and mapset are set */
     G_location_path();
     mapset = G_mapset();
@@ -83,11 +83,12 @@ void G__no_gisinit(const char *version)
     if (initialized)
 	return;
 
-    if (strcmp(version, GIS_H_VERSION) != 0)
+    /*if (strcmp(version, GIS_H_VERSION) != 0)
 	G_fatal_error(_("Module built against version %s but "
 			"trying to use version %s. "
 			"You need to rebuild GRASS GIS or untangle multiple installations."),
                         version, GIS_H_VERSION);
+    */
     gisinit();
 }
 
